src/mixins/object_interactivity.mixin.ts

```diff
@@ -1,394 +1,497 @@
+    for (const i in this.controls) {
+      fn(this.controls[i], i, this);
+    }
+  }
+
+  /**
+   * Sets the coordinates that determine the interaction area of each control
+   * note: if we would switch to ROUND corner area, all of this would disappear.
+   * everything would resolve to a single point and a pythagorean theorem for the distance
+   * @todo evaluate simplification of code switching to circle interaction area at runtime
+   * @private
+   */
+  _setCornerCoords(): void {
+    Object.entries(this.oCoords).forEach(([controlKey, control]) => {
+      const controlObject = this.controls[controlKey];
+      control.corner = controlObject.calcCornerCoords(
+        this.angle,
+        this.cornerSize,
+        control.x,
+        control.y,
+        false
+      );
+      control.touchCorner = controlObject.calcCornerCoords(
+        this.angle,
+        this.touchCornerSize,
+        control.x,
+        control.y,
+        true
+      );
+    });
+  }
+
+  /**
+   * Draws a colored layer behind the object, inside its selection borders.
+   * Requires public options: padding, selectionBackgroundColor
+   * this function is called when the context is transformed
+   * has checks to be skipped when the object is on a staticCanvas
+   * @todo evaluate if make this disappear in favor of a pre-render hook for objects
+   * this was added by Andrea Bogazzi to make possible some feature for work reasons
+   * it seemed a good option, now is an edge case
+   * @param {CanvasRenderingContext2D} ctx Context to draw on
+   */
+  drawSelectionBackground(ctx: CanvasRenderingContext2D): void {
+    if (
+      !this.selectionBackgroundColor ||
+      (this.canvas && !this.canvas.interactive) ||
+      (this.canvas && this.canvas._activeObject !== this)
+    ) {
+      return;
+    }
+    ctx.save();
+    const center = this.getRelativeCenterPoint(),
+      wh = this._calculateCurrentDimensions(),
+      vpt = this.canvas.viewportTransform;
+    ctx.translate(center.x, center.y);
+    ctx.scale(1 / vpt[0], 1 / vpt[3]);
+    ctx.rotate(degreesToRadians(this.angle));
+    ctx.fillStyle = this.selectionBackgroundColor;
+    ctx.fillRect(-wh.x / 2, -wh.y / 2, wh.x, wh.y);
+    ctx.restore();
+  }
+
+  /**
+   * @public override this function in order to customize the drawing of the control box, e.g. rounded corners, different border style.
+   * @param {CanvasRenderingContext2D} ctx ctx is rotated and translated so that (0,0) is at object's center
+   * @param {Point} size the control box size used
+   */
+  strokeBorders(ctx: CanvasRenderingContext2D, size: Point): void {
+    ctx.strokeRect(-size.x / 2, -size.y / 2, size.x, size.y);
+  }
+
+  /**
+   * @private
+   * @param {CanvasRenderingContext2D} ctx Context to draw on
+   * @param {Point} size
+   * @param {Object} styleOverride object to override the object style
+   */
+  _drawBorders(ctx: CanvasRenderingContext2D, size: Point, styleOverride: Record<string, any> = {}): void {
```

Comment:
private?

Reply:
maybe? not sure, at some point we have to decide what becomes private for the sake of reducing exposure Not sure if someone would want to call _drawBorders directly